src/main/java/edu/ithaca/dragon/blackjack/Deck.java

```diff
@@ -24,4 +26,37 @@ public Deck(){
     public ArrayList<Card> getNotInDeck() {
         return notInDeck;
     }
+
+    public Card getNextCard() throws NoMoreCardsException{
+        if(deck.size() <= 0)throw new NoMoreCardsException("No More Cards");
+        Card rtn = deck.get(0);
+        notInDeck.add(rtn);
+        deck.remove(0);
+        return rtn;
+    }
+
+    public void shuffle(){
+        if(notInDeck.size() > 0) {
+            for (int x = 0; x < notInDeck.size(); x++) {
```

Comment:
Everywhere that you write a for loop that just iterates an entire collection, you should use a for each loop, e.g., 
for (Card card: notInDeck) {
    deck.add(card);
}

Reply:
Replaced with a .addAll